com.ibm.wala.dalvik.test/source/com/ibm/wala/dalvik/test/cha/MultiDexScopeTest.java

```diff
@@ -83,17 +84,59 @@ public void testMultiDex() throws ClassHierarchyException, IOException {
         ClassHierarchy cha, cha2;
         String multidexApk =  "data/multidex-test.apk";
 
-        scope = setUpTestScope(multidexApk,"", MultiDexScopeTest.class.getClassLoader());
+        scope = AnalysisScopeReader.readJavaScope("primordial.txt", new File(""), MultiDexScopeTest.class.getClassLoader());
+        scope.setLoaderImpl(ClassLoaderReference.Application,
+                "com.ibm.wala.dalvik.classLoader.WDexClassLoaderImpl");
+
+        try {
+            // extract dex files to disk and add them manually to scope
+            File dexTmpDir = new File(System.getProperty("java.io.tmpdir"));
+            extractDexFiles(multidexApk, dexTmpDir);
+
+            File dex1 = new File(dexTmpDir + File.separator + "classes.dex");
+            scope.addToScope(ClassLoaderReference.Application, DexFileModule.make(dex1));
+            dex1.delete();
+
+            File dex2 = new File(dexTmpDir + File.separator + "classes2.dex");
+            scope.addToScope(ClassLoaderReference.Application, DexFileModule.make(dex2));
+            dex2.delete();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
         cha = ClassHierarchyFactory.make(scope);
 
+        // use setUpAndroidAnalysisScope
         scope2 = Util.makeDalvikScope(null,null, multidexApk);
         cha2 = ClassHierarchyFactory.make(scope2);
 
         Assert.assertEquals(Integer.valueOf(getNumberOfAppClasses(cha)),Integer.valueOf(5));
-        Assert.assertNotEquals(Integer.valueOf(getNumberOfAppClasses(cha)), Integer.valueOf(getNumberOfAppClasses(cha2)));
+        Assert.assertEquals(Integer.valueOf(getNumberOfAppClasses(cha)), Integer.valueOf(getNumberOfAppClasses(cha2)));
 
     }
 
+    private static void extractDexFiles(String apkFileName, File outDir) throws IOException {
+        ZipInputStream zis = new ZipInputStream(new FileInputStream(new File(apkFileName)));
```

Comment:
The Maven build fails since you don't use try-with-resources for these

Reply:
Hmm, it seems that the maven build checks for additional things compared to the gradle build. I only use the gradle build and it did not show this.

Reply:
Yes, it's a bit annoying.  I have a vague plan to get rid of the Maven checks and switch the Gradle builds to Error Prone /cc @liblit

Reply:
I believe (but am not absolutely certain) that Maven builds use the Eclipse JDT core batch compiler (`ecj`), whereas Gradle builds use the standard JDK compiler (`javac`). I also believe (but am even less certain) that the Maven+`ecj` builds use the Eclipse project configuration’s settings for how errors and warnings should be treated, which have intentionally been cranked up to be quite pedantic in places.

WALA’s Travis CI tests use both build systems, which means you get two different compilers’ opinions as to the validity of your code. That can be confusing if you’re not expecting it, but also has some benefits. However, if this is seen as more trouble than it is worth, we could consider using the [ECJ Plugin for Gradle](https://github.com/xinthink/gradle-ecj) to have Gradle use `ecj` as well.